docs/user-guide/deployments.md

```diff
@@ -0,0 +1,414 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/user-guide/deployments.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Deployments
+
+**Table of Contents**
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Deployments](#deployments)
+  - [What is a _Deployment_?](#what-is-a-deployment)
+  - [Enabling Deployments on kubernetes cluster](#enabling-deployments-on-kubernetes-cluster)
+  - [Creating a Deployment](#creating-a-deployment)
+  - [Updating a Deployment](#updating-a-deployment)
+    - [Multiple Updates](#multiple-updates)
+  - [Writing a Deployment Spec](#writing-a-deployment-spec)
+    - [Pod Template](#pod-template)
+    - [Replicas](#replicas)
+    - [Selector](#selector)
+    - [Unique Label Key](#unique-label-key)
+    - [Strategy](#strategy)
+      - [Recreate Deployment](#recreate-deployment)
+      - [Rolling Update Deployment](#rolling-update-deployment)
+        - [Max Unavailable](#max-unavailable)
+        - [Max Surge](#max-surge)
+        - [Min Ready Seconds](#min-ready-seconds)
+  - [Alternative to Deployments](#alternative-to-deployments)
+    - [kubectl rolling update](#kubectl-rolling-update)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## What is a _Deployment_?
+
+A _Deployment_ provides declarative update for Pods and ReplicationControllers.
+Users describe the desired state in deployment object and deployment
+controller changes the actual state to that at a controlled rate.
+Users can define deployments to create new resources, or replace existing ones
+by new ones.
+
+A typical use case is:
+* Create a deployment to bring up a replication controller and pods.
+* Later, update that deployment to recreate the pods (for ex: to use a new image).
+
+## Enabling Deployments on kubernetes cluster
+
+Deployments is part of the [`extensions` API Group](../api.md#api-groups) and is not enabled by default.
+Set `--runtime-config=extensions/v1beta1/deployments=true` on API server to
+enable it.
+This can be achieved by exporting `ENABLE_DEPLOYMENTS=true` before running
+`kube-up.sh` script on GCE.
+
+Note that Deployment objects effectively have [API version
+`v1alpha1`](../api.md#api-versioning).
+Alpha objects may change or even be discontinued in future software releases.
+However, due to to a known issue, they will appear as API version `v1beta1` if
+enabled.
+
+## Creating a Deployment
+
+Here is an example Deployment. It creates a replication controller to
+bring up 3 nginx pods.
+
+<!-- BEGIN MUNGE: EXAMPLE nginx-deployment.yaml -->
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx-deployment
```

Comment:
Is there a reason why these labels are different from those on the pods? My normal advice would be to let the labels and selector be set by default from the labels in the pod template.